js/utilities.js

```diff
@@ -140,6 +140,154 @@ var ERMrest = (function(module) {
+     * @function
+     * @param {Object} value A boolean value to transform
+     * @param {Object} [options] Configuration options
+     * @return {string} A string representation of a boolean value
+     * @desc Formats a given boolean value into a string for display
+     */
+    module._printBoolean = function(value, options) {
+        options = (typeof options === 'undefined') ? {} : options;
+        if (value === null) {
+            return '';
+        }
+        // TODO: What kinds of options are we supporting?
+        return Boolean(value).toString();
+    }
+
+    /**
+     * @function
+     * @param {Object} value An integer value to transform
+     * @param {Object} [options] Configuration options
+     * @return {string} A string representation of value
+     * @desc Formats a given integer value into a whole number (with a thousands
+     * separator if necessary), which is transformed into a string for display.
+     */
+    module._printInteger = function(value, options) {
+        options = (typeof options === 'undefined') ? {} : options;
+        if (value === null) {
+            return '';
+        }
+
+        // Remove fractional digits; implicitly transform value to number type
+        // TODO: Truncate or round?
+        value = Math.trunc(value);
+
+        // Add comma separators
+        return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+    }
+
+    /**
+     * @function
+     * @param {Object} value An timestamp value to transform
+     * @param {Object} [options] Configuration options
+     * @return {string} A string representation of value. Default is ISO string.
+     * @desc Formats a given timestamp value into a string for display.
+     */
+    module._printTimestamp = function(value, options) {
+        options = (typeof options === 'undefined') ? {} : options;
+        if (value === null) {
+            return '';
+        }
+        var year, month, date, hour, minute, second, ms;
+        try {
+            value = value.toString();
+            value = new Date(value);
+            // Later when we support more formats, we'll probably need to manually
+            // construct the date time with the following pieces:
+
+            // year = value.getFullYear();
+            // month = value.getMonth() + 1;
+            // date = value.getDate();
+            // hour = value.getHours();
+            // minute = value.getMinutes();
+            // second = value.getSeconds();
+            // ms = value.getMilliseconds();
+        } catch (exception) {
+            // Is this the right error?
+            throw new module.InvalidInputError("Couldn't extract timestamp from input" + exception);
+        }
+
+        if (typeof value.getTime() !== 'number') {
+            // Invalid timestamp
+            throw new module.InvalidInputError("Couldn't transform input to a valid timestamp");
+        }
+
+        // ISOString format: YYYY-MM-DDTHH:mm:ss.sssZ
+        return value.toISOString();
```

Comment:
Heh, timestamps from ERMrest will be in ISO formatted strings like that already except usually with a timezone offset like `-0700` rather than `Z`.  Most people find them difficult to read if you don't at least replace the 'T' with a space.

I think that a useful formatting option would need to provide localization choices and/or access to a structured timestamp representation so that custom formatting strings in an annotation could alter presentation of a column, whether to truncate precision, reorder fields, or change separators.